staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -356,8 +356,19 @@ func (o LogsOptions) Validate() error {
 	return nil
 }
 
-// RunLogs retrieves a pod log
+// RunLogs wraps RunLogsContext with signal handling.
+// When a signal is received, streaming is stopped, then followed by os.Exit(1).
 func (o LogsOptions) RunLogs() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	intr := interrupt.New(nil, cancel)
+	return intr.Run(func() error {
+		return o.RunLogsContext(ctx)
+	})
+}
+
+// RunLogsContext retrieves a pod log.
```

Comment:
So this function is exported. That means it can be used by other applications. In my opinion, it is better to enrich the description in here to set the expectations (as you did in RunLogs). For example, RunLogsContext does not have interrupt handlers which means that signals won't be captured. That means it is up to the caller.